### PR TITLE
Income tests

### PR DESCRIPTION
**Pull request by Stanislavcod**

### CODE REVIEW

File: ExpenseMaster.BusinessLogic/Implementations/IncomeService.cs [hunk 1]
```diff
@@ -28,7 +28,11 @@ public async Task<IEnumerable<IncomeItemDto>> GetAllIncomes()
         public async Task<IncomeItemDto> GetIncomeById(int id)
         {
             var result = await _repositoryWrapper.Income.FindByConditionAsync(x => x.Id == id);
-            var income = await result.FirstOrDefaultAsync();
+            var income = result.FirstOrDefault();
```
Comment (KatsiarynaAnashkevich) [inline ExpenseMaster.BusinessLogic/Implementations/IncomeService.cs:31]: Why do you change it? It is interesting for me) (p.s. you do not have to fix it)

File: ExpenseMaster.Tests/Services/BudgetServiceTests.cs [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+            var mockMapper = new Mock<IMapper>();
+
+            mockRepositoryWrapper
+                .Setup(r => r.Budget.GetByIdAsync(budgetId))
+                .ReturnsAsync((Budget)null);
+
+            var budgetService = new BudgetService(mockRepositoryWrapper.Object, mockMapper.Object);
+
+            // Act and Assert
+            await Assert.ThrowsAsync<InvalidOperationException>(() => budgetService.GetByIdAsync(budgetId));
```
Comment (KatsiarynaAnashkevich) [inline ExpenseMaster.Tests/Services/BudgetServiceTests.cs:57]: separate them

File: ExpenseMaster.Tests/Repository/BudgetRepositoryTests.cs [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+using Microsoft.EntityFrameworkCore;
+using Moq;
+using Xunit;
+
+namespace ExpenseMaster.Tests.Repository
+{
+    public class BudgetRepositoryTests
+    {
+        [Fact]
+        public async Task GetByIdAsync_ReturnsBudget()
```
Comment (KatsiarynaAnashkevich) [inline ExpenseMaster.Tests/Repository/BudgetRepositoryTests.cs:13]: naming

File: ExpenseMaster.Tests/Repository/BudgetRepositoryTests.cs [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+            };
+            var mockDbSet = GetMockDbSet(budgets);
+            var mockContext = new Mock<ApplicationDatabaseContext>();
+            mockContext.Setup(c => c.Set<Budget>()).Returns(mockDbSet.Object);
+
+            var repository = new BudgetRepository(mockContext.Object);
+
+            var result = await repository.GetByIdAsync(2);
+
+            Assert.NotNull(result);
```
Comment (KatsiarynaAnashkevich) [inline ExpenseMaster.Tests/Repository/BudgetRepositoryTests.cs:29]: fluent assertion

File: ExpenseMaster.Tests/Services/IncomeServiceTests.cs [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+using ExpenseMaster.DAL.Models;
+using Moq;
+using System.Linq.Expressions;
+using Xunit;
+
+namespace ExpenseMaster.Tests.Services
+{
+    public class IncomeServiceTests
+    {
+        private readonly Mock<IRepositoryWrapper> _mockRepositoryWrapper;
```
Comment (KatsiarynaAnashkevich) [inline ExpenseMaster.Tests/Services/IncomeServiceTests.cs:14]: Why do you write like here only in this file? Why others like this
image

File: ExpenseMaster.Tests/Services/BudgetServiceTests.cs (hunk 2 above)
Comment (Stanislavcod) [reply ExpenseMaster.Tests/Services/BudgetServiceTests.cs:57]: it's not my code and it's old

Comment (Stanislavcod) [reply ExpenseMaster.Tests/Services/BudgetServiceTests.cs:57]: it's not my code and it's old

File: ExpenseMaster.Tests/Repository/BudgetRepositoryTests.cs (hunk 3 above)
Comment (Stanislavcod) [reply ExpenseMaster.Tests/Repository/BudgetRepositoryTests.cs:13]: it's not my code and it's old

**Pull request merged**